pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -363,10 +363,10 @@ func (fw *FileWriter) Remove() error {
 	return os.Remove(fw.name)
 }
 
-func (w *binaryWriter) AddIndexMeta(indexVersion int, indexPostingOffsetTable uint64) error {
+func (w *binaryWriter) AddIndexMeta(indexVersion int, indexLabelOffsetTableOffset uint64) error {
```

Comment:
sooo I considered this initially, but my concern was that this isn't the end of the last posting list. It's the end of the last posting list __+1__. I was concerned about an off-by-one error in the future if someone relies on this. WDYT about this, maybe I'm overly worried?

Reply:
What if we were to store the end of the last posting list?

Reply:
so i revisited this. Since our end offsets are exclusive, I think it's ok to have this as `indexEndLastPostingList` with the same value. I only renamed the variables.